### PR TITLE
Change RET_WRONG_IMPLID() to return RMW_RET_INCORRECT_IMPLEMENTATION

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says. Split from #223.

### CODE REVIEW

**Comment by hidmic** [comment]
CI up to `test_rmw_implementation` and `rcl`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12058/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7081/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9831/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11991/)

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 1]
```diff
@@ -2780,6 +2792,7 @@ extern "C" rmw_ret_t rmw_publisher_event_init(
 extern "C" rmw_ret_t rmw_subscription_event_init(
   rmw_event_t * rmw_event, const rmw_subscription_t * subscription, rmw_event_type_t event_type)
 {
+  RET_NULL(subscription);
   RET_WRONG_IMPLID_X(subscription, return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
```
Comment (ivanpauno) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:2796]: can be replaced with RET_WRONG_IMPLID?

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 2]
```diff
@@ -2769,6 +2780,7 @@ static rmw_ret_t init_rmw_event(
 extern "C" rmw_ret_t rmw_publisher_event_init(
   rmw_event_t * rmw_event, const rmw_publisher_t * publisher, rmw_event_type_t event_type)
 {
+  RET_NULL(publisher);
   RET_WRONG_IMPLID_X(publisher, return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
```
Comment (ivanpauno) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:2784]: here too

File: rmw_cyclonedds_cpp/src/rmw_node.cpp (hunk 1 above)
Comment (hidmic) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp]: Indeed. See  https://github.com/ros2/rmw_cyclonedds/pull/226/commits/4432c12c86be2d44ad7f57249339d2756bdbe98d.

File: rmw_cyclonedds_cpp/src/rmw_node.cpp (hunk 2 above)
Comment (hidmic) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp]: Yeap! See https://github.com/ros2/rmw_cyclonedds/pull/226/commits/4432c12c86be2d44ad7f57249339d2756bdbe98d.

**Comment by hidmic** [comment]
Alright CI's green, PR job's green, stable Github actions are green. Merging!

**Pull request merged**